### PR TITLE
AC-1710: add deployment status to detailed and list view endpoints

### PR DESCRIPTION
**Pull request by samrooke**
### Description
Add deployment parameters to each application version on the list and detailed view requests.

Fixes # .

A few sentences describing the changes proposed in this pull request.

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] All tests passed locally.

### CODE REVIEW

File: src/modules/store/applications/applications.module.ts [hunk 1]
```diff
@@ -19,18 +19,14 @@ import { AuthModule } from './../auth/auth.module';
...
 import { AppInstallation } from 'shared/typeorm/entities/app-installation.entity';
 
 @Module({
   imports: [
     TypeOrmModule.forFeature([AppInstallation]),
-    HttpModule.registerAsync({
-      useClass: HttpAideConfigService,
-    }),
     AuthModule,
+    TypeOrmModule.forFeature([AppInstallation]),
```
[the remark this answers is not in the document]
Comment (samrooke) [reply src/modules/store/applications/applications.module.ts:29]: Good spot - I don't think the rebase made these conflict so I didn't see it

File: src/modules/admin/templates/templates.service.spec.ts [hunk 2]
```diff
@@ -175,6 +183,9 @@ describe('TemplatesService', () => {
       httpService.get.mockReturnValue(makeObservableForTest(axios.get));
```
Comment (mehranrashid) [inline src/modules/admin/templates/templates.service.spec.ts:183]: I don't even know how the unit tests are passing when the GET request has been removed and we're expecting the method to have been called on line 212, unless I'm missing something?

File: src/modules/store/medical-specialties/medical-specialties.service.ts [hunk 3]
```diff
@@ -33,11 +34,14 @@ export class MedicalSpecialtiesService {
       .getAuthToken()
       .then(async (token) => {
         const appResponse = await lastValueFrom(
-          this.httpService.get<any>(`/app-store/medical-specialties`, {
-            headers: {
-              Authorization: 'Bearer ' + token.access_token,
+          this.httpService.get<IMedicalSpecialty[]>(
+            `/app-store/medical-specialties/`,
```
Comment (mehranrashid) [inline src/modules/store/medical-specialties/medical-specialties.service.ts:38]: Will the `/` at the end cause a problem?

Comment (DanielNuttall-AD) [reply src/modules/store/medical-specialties/medical-specialties.service.ts:38]: shouldn't make a difference :)

**Review by DanielNuttall-AD — APPROVED** [review]

File: src/modules/store/medical-specialties/medical-specialties.service.ts (hunk 3 above)
Comment (samrooke) [reply src/modules/store/medical-specialties/medical-specialties.service.ts:38]: I only changed the interface fyi, but yeah the end slash won't have any effect

File: src/modules/admin/templates/templates.service.spec.ts (hunk 2 above)
Comment (samrooke) [reply src/modules/admin/templates/templates.service.spec.ts:183]: It is technically still being called by the app service. I think Rowell defined a method in this class to get the app details but we already had that method in the app service so it's just doing it through that instead

**Review by rowellx68 — APPROVED** [review]

File: src/modules/store/applications/applications.service.ts [hunk 4]
```diff
@@ -61,37 +64,75 @@ export class ApplicationsService {
...
+            where: {
+              application_id: app.id,
+              status: WorkflowStatus.INSTALLED,
+            },
+            order: {
+              installation_date: 'DESC',
+            },
+          });
+
+          updatedApp.latest_deployed_version_string =
```
Comment (samrooke) [inline src/modules/store/applications/applications.service.ts:91]: This has been changed from a list of version strings to the latest deployed version string.

**Review by DanielNuttall-AD — APPROVED** [review]

**Review by mehranrashid — APPROVED** [review]

**Pull request merged**